src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -148,7 +148,9 @@ impl Step for ToolBuild {
             &self.extra_features,
         );
 
-        if path.ends_with("/rustdoc") &&
+        // Rustc tools (miri, clippy, cargo, rustfmt, rust-analyzer)
+        // could use the additional optimizations.
+        if self.mode == Mode::ToolRustc &&
             // rustdoc is performance sensitive, so apply LTO to it.
```

Comment:
I have no object to the change, but the new comment should probably merge in this one.

Reply:
Oops, good catch! https://github.com/rust-lang/rust/pull/139707